### ISSUE TITLE
Examples are rebuilt again with every run of 'make' in the 'example' directory

### ISSUE THREAD

**Issue by yurivict** [Immediate-Mode-UI/Nuklear#415]

**Comment by kiner-shah** [comment]
Yes because the Makefile seems to have rules which forces cleaning of `bin` directory:
```
all: generate file_browser extended canvas skinning

generate: clean
ifeq ($(OS),Windows_NT)
	@mkdir bin 2> nul || exit 0
else
	@mkdir -p bin	
endif

clean:
	$(DEL_BIN)
```
`all` requires `generate` which requires `clean`.

**Comment by yurivict** [comment]
```clean``` shouldn't be part of ```all```.

**Comment by kiner-shah** [comment]
> `clean` shouldn't be part of `all`.

Yes, not sure why they added `clean` as a part of `all`. You may fix it and submit a pull request here.

**Comment by sleeptightAnsiC** [comment]
Hey,

The current Makefiles all behave this way because they don’t support incremental compilation, meaning they're unable to detect whether you modified any file or not. Cleaning and rebuilding every time is just a hacky workaround to this problem.

I've recently contributed a new demo with modified Makefile. This one is able to detect the changes and rebuilds only when necessary, see https://github.com/Immediate-Mode-UI/Nuklear/blob/9afb3dd6d14fba0c5e62f08809b494b6d580b9b1/demo/sdl3_renderer/Makefile

Do you think this would be enough?

Personally, I don't like the fact that the demos are all using the amalgamated header (`Nuklear/nuklear.h`) instead of building the library from source (`Nuklear/src/*`), but I guess, changing this could annoy a lot of people.
